python/lsst/drp/tasks/assemble_cell_coadd.py

```diff
@@ -0,0 +1,372 @@
+from lsst.pipe.base import NoWorkFound, PipelineTask, PipelineTaskConfig, PipelineTaskConnections, Struct
+from lsst.pipe.base.connectionTypes import Input, Output
+from lsst.pipe.tasks.coaddBase import makeSkyInfo
+from lsst.pipe.tasks.coaddInputRecorder import CoaddInputRecorderTask
+from lsst.pipe.tasks.interpImage import InterpImageTask
+from lsst.pipe.tasks.scaleZeroPoint import ScaleZeroPointTask
+from lsst.skymap import BaseSkyMap
+
+
+class AssembleCellCoaddConnections(
+    PipelineTaskConnections,
+    dimensions=("tract", "patch", "band", "skymap"),
+    defaultTemplates={"inputWarpName": "deep", "outputCoaddSuffix": "Cell"},
+):
+    inputWarps = Input(
+        doc="Input warps",
+        name="{inputWarpName}Coadd_directWarp",
+        storageClass="ExposureF",
+        dimensions=("tract", "patch", "skymap", "visit", "instrument"),
+        deferLoad=True,
+        multiple=True,
+    )
+
+    skyMap = Input(
+        doc="Input definition of geometry/bbox and projection/wcs",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+
+    multipleCellCoadd = Output(
+        doc="Output multiple cell coadd",
+        name="{inputWarpName}Coadd{outputCoaddSuffix}",
+        storageClass="MultipleCellCoadd",
+        dimensions=("tract", "patch", "band", "skymap"),
+    )
+
+
+class AssembleCellCoaddConfig(PipelineTaskConfig, pipelineConnections=AssembleCellCoaddConnections):
+    doInterp = Field[bool](doc="Interpolate over NaN pixels?", default=False)
+    interpImage = ConfigurableField(
+        target=InterpImageTask,
+        doc="Task to interpolate images (and extrapolate) over NaN pixels",
+    )
+    scaleZeroPoint = ConfigurableField(
+        target=ScaleZeroPointTask,
+        doc="Task to scale images to a common zero point",
+    )
+    badMaskPlanes = ListField[str](
+        doc="Mask planes that, if set, the corresponding pixel would not be included in the coadd.",
+        default=("BAD", "EDGE", "NO_DATA", "SAT"),
+    )
+    calcErrorFromInputVariance = Field[bool](
+        doc="Calculate coadd variance from input variance by stacking "
+        "statistic. Passed to "
+        "StatisticsControl.setCalcErrorFromInputVariance()",
+        default=True,
+    )
+    max_maskfrac = RangeField[float](
+        doc="Maximum fraction of masked pixels in a cell",
+        default=0.99,
+        min=0.0,
+        max=1.0,
+        inclusiveMin=True,
+        inclusiveMax=False,
+    )
+    # The following config options are specific to the CoaddPsf.
+    coaddPsf = ConfigField(
+        doc="Configuration for CoaddPsf",
+        dtype=CoaddPsfConfig,
+    )
+    inputRecorder = ConfigurableField(
+        doc="Subtask that helps fill CoaddInputs catalogs added to the final Exposure",
+        target=CoaddInputRecorderTask,
+    )
+
+
+class AssembleCellCoaddTask(PipelineTask):
```

Comment:
When you add docstrings for this class, let's add a note that says something to the effect of, "this task and especially its configs are still experimental and may change in the future without RFC", because I think there are some things we may want to change after this ticket but before this is included in a production pipeline, and I'd rather not have to go through RFCs and deprecations for those.

Reply:
Yes. And I'd add that it's subject to change until it is included in the standard DRP pipeline(s).